ext/bg/js/backend.js

```diff
@@ -137,7 +143,26 @@ class Backend {
         ]);
     }
 
-    async prepare() {
+    prepare() {
+        if (this._preparePromise === null) {
+            const promise = this._prepareInternal();
+            promise.then(
+                (value) => {
+                    this._isPrepared = true;
+                    this._prepareCompleteResolve(value);
+                },
+                (error) => {
+                    this._prepareError = true;
+                    this._prepareCompleteReject(error);
+                }
+            );
+            promise.finally(() => this._updateBadge());
+            this._preparePromise = promise;
+        }
+        return this._prepareCompletePromise;
```

Comment:
It appears that this is not used (`background-main.js` doesn't await); is this intentional?

Reply:
The intent is to have it match a call to `.prepareComplete()` in terms of what is returned. So it is intentional, but current functionality doesn't use it. The reason `await` was removed from background-main.js is because there was nothing after it, and having the entry point function be non-`async` makes the semantics more clear. I.e. for anyone who isn't perfectly familiar with how/when promises invoke, it will be clear that there are no deferrals before `BackendEventHandler` is prepared.

Reply:
Ok, in that case the PR looks good to me.